src/frontend/src/session.rs

```diff
@@ -259,12 +262,21 @@ impl FrontendEnv {
             .unwrap();
         info!("advertise addr is {}", frontend_address);
 
+        let rpc_addr: HostAddr = opts.frontend_rpc_listener_addr.parse().unwrap();
+        let frontend_rpc_addr = HostAddr {
+            // Use the host of advertise address for the frontend rpc address.
+            host: frontend_address.host.clone(),
+            port: rpc_addr.port,
+        };
         // Register in meta by calling `AddWorkerNode` RPC.
         let (meta_client, system_params_reader) = MetaClient::register_new(
             opts.meta_addr,
             WorkerType::Frontend,
             &frontend_address,
-            Default::default(),
+            AddWorkerNodeProperty {
+                secondary_host: frontend_rpc_addr.to_string(),
```

Comment:
Refactored to store the frontend rpc addr as a property according to last week‘s discussion with @yezizp2012. cc @BugenZhao

Reply:
Can you elaborate more on the motivation?

Reply:
We want to retain the advertise addr (port 4566) for frontend to display in the UI, e.g. in the cluster info or kernel dashboard, since it makes more sense to show up this address.

In theory, we can share the advertise addr for psql and grpc service, since both of them are application layer protocol. If we only need to support tcp for psql, we can pass the `TcpListener` to tonic to share the same port. But we also have unix socket protocol for psql.